### DISCUSSION TITLE
inconsistent "Always Fit Window" behavior in VLC for two different videos, one processed with MLT, the other with FFMPEG

### DISCUSSION THREAD

**Discussion by xmatic** [mltframework/mlt#1055] [Q&A] [answered]
Two different videos: 

 - both have same resolution (1920 x 1080)
 - one processed with `Tencoder` (frontend for FFMPEG), and the other processed with `kdenlive` (video editor that uses MLT framework)

**The issue:** they show up very differently in VLC. 

 - One needs "Menu > Video > Always Fit Window" to be **checked** to fill the screen.
 - The other needs "Menu > Video > Always Fit Window" to be **unchecked** to fill the screen.

----

**Video 1**: processed using `kdenlive` (video editor that uses MLT framework) [link][1]:

How the file looks in the file system:

![image](https://github.com/user-attachments/assets/15e35aeb-7fe6-47a1-ae3e-8878e44630e7)

`kdenlive` project settings:

![image](https://github.com/user-attachments/assets/ff822075-2226-466b-a355-62f0dc924328)

`kdenlive` render profile:

![image](https://github.com/user-attachments/assets/26821523-b42d-4bf1-8902-29714e194fb9)

The MLT command looks like this in kdenlive (screenshot):

![2024-11-26_002205](https://github.com/user-attachments/assets/6c2718d8-8c3a-4796-8d9f-798948193966)

The MLT command in text form: 

`ab=160k acodec=aac channels=2 crf=23 f=matroska g=15 movflags=+faststart preset=veryfast real_time=-1 threads=0 vcodec=libx264`

MediaInfo:

![image](https://github.com/user-attachments/assets/99a18d9b-21a3-4945-a09b-1b193310a0cb)

VLC with "Always Fit Window" **checked**: video underfills the screen:

![image](https://github.com/user-attachments/assets/429dde4c-eb1d-490e-abfe-59af3c8a3682)

VLC with "Always Fit Window" **unchecked**: video fills the screen:

![image](https://github.com/user-attachments/assets/b376c98b-0851-4d96-8757-26835fb509b9)

----

**Video 2**: processed using `Tencoder` (frontend for FFMPEG):

How the file looks in the file system:

![image](https://github.com/user-attachments/assets/acae4d66-38a7-488b-a033-982db15e5306)

`Tencoder` encode settings:

![image](https://github.com/user-attachments/assets/e8f47304-d8b6-4081-8319-f8402151a38b)

MediaInfo:

![image](https://github.com/user-attachments/assets/d46c9430-3340-4e25-b4b6-b314550229a0)

VLC with "Always Fit Window" **checked**: video fills the screen:

![image](https://github.com/user-attachments/assets/a8d8e34f-6d5c-417c-89f5-0322eda75506)

VLC with "Always Fit Window" **unchecked**: video overfills the screen:

![image](https://github.com/user-attachments/assets/882b9cf6-f609-4f14-8adb-0f2d57f6399a)

----

Summary:

 - despite the two videos having the same resolution (1920 * 1080)
 - to fill the screen, one video needs "Always Fit Window" **unchecked** and the other needs it **checked**.
 
Question for you dear reader: **What can I do in kdenlive or MLT so that the video fills the screen with "Always Fit Window" checked?**

**Comment by ddennedy** [comment, marked as answer]
In Kdenlive project settings, the pixel aspect ratio and display aspect ratio are incongruent with the resolution: they do not mathematically make sense. Usually, only one pair of aspect ratio fields are provided to avoid this problem. I do not know why Kdenlive chooses this UI. 


The MediaInfo for your Tencoder output shows an aspect ratio of 0.562. This is equivalent to a 9:16 aspect ratio vertical video. Probably in Kdenlive you want to use:
Resolution: 1080 x 1920  (notice the order is swapped from what you have)
Pixel aspect ratio: 1 / 1
Display aspect ratio: 9 / 16

This is actually more suitable for the Kdenlive discussion forum. MLT is primarily for developers, and Kdenlive is the app you are using that is responsible for how it uses MLT.

**Comment by bmatherly** [comment]
It looks like you are trying to make a vertical video. But your export settings do not form a vertical video:
![image](https://github.com/user-attachments/assets/df85b735-b57e-4ec0-96d4-ff3705fb19c0)

I would suggest:
Size: 1080x1920
Pixel Aspect Ratio: 1x1
Display Aspect Ratio: 9x16

I am not a Kdenlive user and there are not any users around here. I recommend that you try the official Kdenlive support forum:
https://discuss.kde.org/tag/kdenlive

**Comment by xmatic** [comment]
[ddennedy](https://github.com/ddennedy), [bmatherly](https://github.com/bmatherly)

Thank you to both of you. Both of you gave the same answer, and it worked.

This is an awesome community. I got the answer so quickly.

Thank you again.
